### PR TITLE
Adding Airplay Support

### PR DESCRIPTION
**Pull request by hghazzi**
***Done***
- Available in Premium edition license or higher
- Airplay icon in the control bar when airplay config is enabled.
- Enabled native Airplay functionality
- Transition to Casting State
- Play, pause, seek, disconnect
- Hide the volume slider while AirPlay is active (currently behaves
like Chromecast)
- Airplay Icon active state
- Disabled Ad playback

JW7-1963

### CODE REVIEW

**Comment by robwalch** [comment]
The PR builders printing cucumber results with TRIGGERED_BUILD_NUMBER - 1 should be updated to included fixes made in the jw7_commercial job.

File: src/js/view/controlbar.js [hunk 1]
```diff
@@ -307,6 +312,15 @@ define([
         onCastActive : function(model, val) {
             utils.toggleClass(this.elements.cast.element(), 'jw-off', !val);
         },
+        onAirplayAvailable : function(model, val) {
+            this.elements.airplay.toggle(val);
+        },
+        onAirplayActive : function(model, val) {
+            utils.toggleClass(this.elements.airplay.element(), 'jw-off', !val);
+            if (this.elements.mute) {
+                this.elements.mute.toggle(!val);
```
Comment (hghazzi) [inline src/js/view/controlbar.js]: Hey @robwalch, I would expect this to hide the volume button, but it doesn't seem to be working. Any ideas?

Comment (robwalch) [inline src/js/view/controlbar.js]: It looks like it should.

**Comment by robwalch** [comment]
test this please

File: src/js/view/view.js [hunk 2]
```diff
@@ -381,6 +382,8 @@ define([
 
             _controlsLayer = _playerElement.getElementsByClassName('jw-controls')[0];
 
+            _controlbarLayer = _playerElement.getElementsByClassName('jw-controlbar')[0];
```
Comment (robwalch) [inline src/js/view/view.js]: Is this used anywhere? If not please remove.

File: src/js/view/controlbar.js [hunk 3]
```diff
@@ -307,6 +312,12 @@ define([
         onCastActive : function(model, val) {
             utils.toggleClass(this.elements.cast.element(), 'jw-off', !val);
         },
+        onAirplayAvailable : function(model, val) {
+            this.elements.airplay.toggle(val);
+        },
+        onAirplayActive : function(model, val) {
+            utils.toggleClass(this.elements.airplay.element(), 'jw-off', !val);
+        },
```
Comment (robwalch) [inline src/js/view/controlbar.js]: Can we move these handlers to commercial?

Comment (hghazzi) [inline src/js/view/controlbar.js]: I would need to extend `controlbar.js` into commercial similarly to how we do it for `view.js`, which we don't have right now. Should I make that change, or leave it as tech debt?

Comment (robwalch) [inline src/js/view/controlbar.js]: OK. We'll leave it as existing tech-debt since we do the same with the cast buttons.

**Comment by alexprice91** [comment]
@robwalch do you have an example of how to enable this via config?

**Comment by robwalch** [comment]
@alexprice91 it will be enabled by default in iOS and macOS Safari once released.

@hghazzi are there any configuration options available for AirPlay like disabling it or player edition requirements?

**Comment by hghazzi** [comment]
Airplay should be enabled by default, like Rob said. It should be only
available for premium edition license or higher, and if you want to disable
Airplay, you add 'enabled': false to the 'cast' config that is currently
being used for chromecast.

**Comment by alexprice91** [comment]
Oh, got it. Any update on release for Enterprise users?

**Comment by robwalch** [comment]
@alexprice91 keep an eye on our [release notes](https://developer.jwplayer.com/jw-player/docs/developer-guide/release_notes/release_notes_7/) for the early beta release. It will not be in our upcoming 7.8 release later this month. We expect to follow up 7.8 with Airplay and Chromecast updates in December.

**Pull request merged**